pkg/kvstore/etcd.go

```diff
@@ -1152,6 +1152,34 @@ func (e *etcdClient) GetIfLocked(ctx context.Context, key string, lock KVLocker)
 	return getR.Kvs[0].Value, nil
 }
 
+// GetWithLeaseInfo returns value of key, together with a boolean indicating if the lease associated
+// with the key is held by this client.
+func (e *etcdClient) GetWithLeaseInfo(ctx context.Context, key string) (bv []byte, hasLease bool, err error) {
```

Comment:
Struggling a bit on how to keep the lease abstraction to the kvstore implementation. The lease manager is only accessible, so returning the actual lease value here would be confusing and painful. Open for other suggestions on how to solve this.

Main thing is that we expect this to "always" be `hasLease == true` - and the case `hasLease == false` or `bv == nil` requires some extra "expensive" synchronization before updating the key.